src/libs/util/readableinterval.cpp

```diff
@@ -0,0 +1,87 @@
+/****************************************************************************
+**
+** Copyright (C) 2015-2018 Oleg Shparber
+** Copyright (C) 2013-2014 Jerzy Kozera
+** Contact: https://go.zealdocs.org/l/contact
+**
+** This file is part of Zeal.
+**
+** Zeal is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** Zeal is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Zeal. If not, see <https://www.gnu.org/licenses/>.
+**
+****************************************************************************/
+
+#include "readableinterval.h"
+
+#include <QStringBuilder>
+#include <QStringList>
+
+using namespace Zeal::Util;
+
+namespace {
+const qint16 SECONDSPERMINUTE = 60;
+const qint16 SECONDSPERHOUR = SECONDSPERMINUTE * 60;
+const qint32 SECONDSPERDAY = SECONDSPERHOUR * 24;
+const qint32 SECONDSPERYEAR = SECONDSPERDAY * 365;
+
+const qint8 MAX_FIELDS_DISPLAYED = 3;
+
+const QString ZERO_INTERVAL_STRING = "now";
```

Comment:
Also, it's better to avoid making translatable strings constant. Just put them inside `tr()` inline.

Reply:
I'm leaving plural forms of units with a `(s)`. Once we apply translation to this project, qt should automatically detect the quantity and append `s`.

http://doc.qt.io/qt-5/i18n-source-translation.html#handling-plurals